internal/builder/builder.go

```diff
@@ -394,6 +367,94 @@ func (b *Builder) Save(logger logging.Logger, creatorMetadata CreatorMetadata) e
 
 // Helpers
 
+func addBuildpacks(logger logging.Logger, tmpDir string, image imgutil.Image, additionalBuildpacks []dist.Buildpack, bpLayers dist.BuildpackLayers) error {
+	type buildpackToAdd struct {
+		tarPath   string
+		diffID    string
+		buildpack dist.Buildpack
+	}
+
+	buildpacksToAdd := map[string]buildpackToAdd{}
+	for i, bp := range additionalBuildpacks {
+		// create buildpack directory
+		bpTmpDir := filepath.Join(tmpDir, strconv.Itoa(i))
+		if err := os.MkdirAll(bpTmpDir, os.ModePerm); err != nil {
+			return errors.Wrap(err, "creating buildpack temp dir")
+		}
+
+		// create tar file
+		bpLayerTar, err := dist.BuildpackToLayerTar(bpTmpDir, bp)
```

Comment:
This additional nested indexed directory prevents multiple buildpacks with the same id + version from colliding based on how `BuildpackToLayerTar` works.